src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
```

Comment:
According to the checklist, nesting loops (indexOf + splice) should be avoided for performance reasons. However, the 'applyDamage' method calls 'Animal.cleanupDeadAnimals' which uses filter, which is the correct approach. But consider moving the cleanup logic to the end of the 'bite' method to ensure that it's only called when necessary, after health has been reduced and potentially reached zero or below.